hub-saml/src/main/java/uk/gov/ida/saml/hub/api/HubTransformersFactory.java

```diff
@@ -333,53 +340,31 @@ public AssertionFromIdpToAssertionTransformer getAssertionFromIdpToAssertionTran
     }
 
     public DecoratedSamlResponseToInboundResponseFromMatchingServiceTransformer getResponseToInboundResponseFromMatchingServiceTransformer(
-            SigningKeyStore signingKeyStore,
-            IdaKeyStore keyStore, String hubEntityId) {
-        return new DecoratedSamlResponseToInboundResponseFromMatchingServiceTransformer(
-                new InboundResponseFromMatchingServiceUnmarshaller(
-                        getAssertionToPassthroughAssertionTransformer(),
-                        new MatchingServiceIdaStatusUnmarshaller()
-                ),
-                getSamlResponseSignatureValidator(getSignatureValidator(signingKeyStore)),
-                this.getSamlResponseAssertionDecrypter(keyStore),
-                getSamlAssertionsSignatureValidator(getSignatureValidator(signingKeyStore)),
-                new EncryptedResponseFromMatchingServiceValidator(),
-                new ResponseAssertionsFromMatchingServiceValidator(
-                        new AssertionValidator(
-                                new IssuerValidator(),
-                                new AssertionSubjectValidator(),
-                                new AssertionAttributeStatementValidator(),
-                                new BasicAssertionSubjectConfirmationValidator()
-                        ),
-                        hubEntityId
-                )
-        );
-    }
-
-    /**
-     * Compliance Tool should implement this method
-     *
-     * @deprecated Compliance Tool should implement this method
-     */
-    @Deprecated
-    public Function<String, InboundResponseFromIdp> getStringToIdaResponseIssuedByIdpTransformer(
             SigningKeyStore signingKeyStore,
             IdaKeyStore keyStore,
-            URI expectedDestinationHost,
-            String expectedEndpoint,
-            IdExpirationCache<String> assertionIdCache,
             String hubEntityId) {
-        // not sure if we need to allow an extra ResponseSizeValidator here.
-        Function<String, Response> t1 = getStringToResponseTransformer();
-        Function<Response, InboundResponseFromIdp> t2 = getDecoratedSamlResponseToIdaResponseIssuedByIdpTransformer(
-                signingKeyStore,
-                keyStore,
-                expectedDestinationHost,
-                expectedEndpoint,
-                assertionIdCache,
-                hubEntityId
+        ResponseAssertionsFromMatchingServiceValidator responseAssertionsFromMatchingServiceValidator = new ResponseAssertionsFromMatchingServiceValidator(
```

Comment:
can't figure out why we've created a instead of working with the existing DecoratedSamlResponseToIdaResponseIssuedByIdpTransformer

Reply:
Not sure about this one, they look completely different to me did you mean the `DecoratedSamlResponseToInboundResponseFromMatchingServiceTransformer`